File: src/backend/Entities/Project.cs
namespace backend.Entities
{
    public enum ProjectStatus
    {
        Proposed,
        InProgress,
        Completed,
        Archived
    }
    public enum Priority
    {
        Low,
        Medium,
        High
    }

    public class Project
    {
        public int Id { get; set; }
        public int? ParentId { get; set; }
        public string ProjectName { get; set; }
        public string Description { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public ProjectStatus ProjectStatus { get; set; }
        public Priority Priority { get; set; }
    }
}

File: src/frontend/src/app/Entities/Project.ts
export interface Project {
  id: number;
  projectName: string;
  description?: string;
  startDate: Date;
  endDate: Date;
  projectStatus: ProjectStatus;
  priority: Priority;
  progress: number;
}

export enum ProjectStatus {
  Proposed,
  InProgress,
  Completed,
  Archived,
}

export enum Priority {
  Low,
  Medium,
  High,
}
